lib/waterline/query/dql/destroy.js

```diff
@@ -88,7 +90,14 @@ module.exports = function(criteria, cb) {
         // than crashing.
         if(!refKey) return next();
 
-        var mappedValues = result.map(function(vals) { return vals[pk]; });
+        // Make sure we don't return any undefined pks
+        var mappedValues = transformedValues.reduce(function(memo, vals) {
+          if (vals[pk]) { 
```

Comment:
What if `vals[pk] === 0`? That would be a valid pk. I think we should replace this line for:
```javascript
 if (vals[pk] !== undefined) { 
```